### PR TITLE
th-377:  Send truck geolocation

### PR DESCRIPTION
**Pull request by albsbz**
Add request for driver geolocation and update geolocation on shift start
Finish geolocation update when shift closed
Restart geolocation update when driver page reloaded and shift open

### CODE REVIEW

**Review by kalinkaaaa14 — APPROVED** [review]

File: frontend/src/libs/packages/middleware/libs/enums/geolocation-messages.enum.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const GeolocationMessages = {
```
Comment (mV22110702) [inline frontend/src/libs/packages/middleware/libs/enums/geolocation-messages.enum.ts:1]:
```suggestion
const GeolocationMessage = {
```

File: frontend/src/slices/trucks/actions.ts [hunk 2]
```diff
@@ -55,6 +55,24 @@ const updateTruckLocationFromSocket = createAsyncThunk<
...
+  },
+);
+
+const restartWatchTruckLocation = createAction(
+  ActionName.RESTART_WATCH_TRUCK_LOCATION,
+  () => {
+    return {
+      payload: null,
+    };
+  },
```
Comment (mV22110702) [inline frontend/src/slices/trucks/actions.ts:73]: An action can be created without a payload, so here it is redundant.

File: frontend/src/libs/packages/middleware/geolocation.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  getState: () => RootReducer;
+}) => {
+  const locationUpdateIntervals: NodeJS.Timer[] = [];
+
+  const setLocationUpdateInterval = (interval: NodeJS.Timer): void => {
+    for (const oldInterval of locationUpdateIntervals) {
+      clearInterval(oldInterval);
+    }
+    locationUpdateIntervals.push(interval);
+  };
```
Comment (mV22110702) [inline frontend/src/libs/packages/middleware/geolocation.middleware.ts:40]: Why are we using array here? Maybe it'd better without it?

```suggestion
  const locationUpdateInterval: NodeJS.Timer|null = null;

  const setLocationUpdateInterval = (interval: NodeJS.Timer): void => {
      locationUpdateInterval = interval;
  };
```

File: frontend/src/libs/packages/middleware/libs/helpers/update-geolocation.helper.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+};
+const handleUpdateError = (error: GeolocationPositionError): void => {
+  getLocationError(error.message);
+};
+
+const updateGeolocation = async (
+  payload: StartWatchTruckLocation,
+  setUpdateLocationInterval: (interval: NodeJS.Timer) => void,
+): Promise<boolean | null> => {
+  if (socketInstance) {
```
Comment (mV22110702) [inline frontend/src/libs/packages/middleware/libs/helpers/update-geolocation.helper.ts:38]: Too big if body. It's better to 
```suggestion
  if (!socketInstance) { return null }
```

**Review by mV22110702 — CHANGES_REQUESTED** [review]

**Review by mV22110702 — APPROVED** [review]

**Pull request merged**